python/tests/test_cli/test_add.py

```diff
@@ -28,9 +28,11 @@ def test_add_github(self) -> None:
             ),
         )
 
-    def test_add_serpapi(self) -> None:
-        """Test `composio add` with no-auth."""
-        self.run("add", "serpapi", input="y")
+    def test_add_auth_mode_auto_uppercase(self) -> None:
+        """Test `composio add` with lowercase --auth-mode."""
```

Comment:
The test docstring could be more descriptive about what it's verifying. Consider adding details about the uppercase conversion and successful execution with the converted value.